src/charm.py

```diff
@@ -1293,25 +1335,16 @@ def _get_runner_scaler(
             ssh_debug_connections=state.ssh_debug_connections,
             repo_policy_compliance=state.charm_config.repo_policy_compliance,
         )
-        # The prefix is set to f"{application_name}-{unit number}"
-        openstack_runner_manager = OpenStackRunnerManager(
-            manager_name=self.app.name,
+        openstack_runner_manager_config = OpenStackRunnerManagerConfig(
+            name=self.app.name,
+            # The prefix is set to f"{application_name}-{unit number}"
             prefix=self.unit.name.replace("/", "-"),
             cloud_config=cloud_config,
             server_config=server_config,
             runner_config=runner_config,
             service_config=service_config,
         )
-        runner_manager_config = RunnerManagerConfig(
-            token=token,
-            path=path,
-        )
-        runner_manager = RunnerManager(
-            manager_name=self.app.name,
-            cloud_runner_manager=openstack_runner_manager,
-            config=runner_manager_config,
-        )
-        return RunnerScaler(runner_manager=runner_manager, reactive_config=state.reactive_config)
+        return openstack_runner_manager_config
```

Comment:
Should this return a runner scaler type? it seems like it is returning a configuration.